vxwhatsapp/consumer.py

```diff
@@ -99,13 +99,21 @@ async def process_message(self, message: IncomingMessage):
             try:
                 await self.submit_message(msg)
             except aiohttp.ClientResponseError as e:
+                # If it's a retryable upstream error, requeue the message
                 if e.status > 499:
                     await message.reject(requeue=True)
+                    # If we've retried this before, wait before retrying again
+                    if message.redelivered:
+                        await sleep(0.5)
```

Comment:
How would this affect throughput if we're retrying a bunch of messages?

Reply:
It would hopefully slow throughput down. If the server is down, we don't want to be retrying as fast as we can, so if we're getting multiple retries is a row, we want to slow down until the server is back up.

Once the server is back up, this code won't be called, so we'll be able to send at normal speed again.

It's not the best method, ideally you want exponential based on number of failures, but since we don't have that here easily, I thought this might be a good way to easily achieve something similar.